citation-server/src/gitasdb/write.js

```diff
@@ -43,12 +49,20 @@ export async function writeObject(type, data) {
 		const objectFields = Object.keys(data);
 		await Promise.all(
 			objectFields.map(async field => {
+				let extension;
+				let writeData;
```

Comment:
This is a first proposition to avoid nested if/else. (probably not the best)
We still have two conditions but now the blocks are differently organised.
For example, we have a schema following this path : 
     if (A) { ... } else { ... }
     if (B) { ... } else { ... }
instead of the previous path : 
     if (A) { if (B) { ... } else { ... } }
     else { if (B) { ...} else { ... }}
which avoid to repeat the nested if we had before.